snapcraft/internal/sources.py

```diff
@@ -86,6 +86,10 @@ def __init__(self, message):
         self.message = message
 
 
+class MissingPackageError(Exception):
```

Comment:
can you consolidate this in the new `snapcraft.errors` and inherit from the SnapcraftError exception in there?